File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-11/14-Poor-Pigs.java
class Solution {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        /*
        states ^ pigs >= buckets
        pigs >= log(buckets) / log(states)
        T: O(1) and S: O(1)
        */
        int states = minutesToTest / minutesToDie + 1;
        return (int) (Math.ceil(Math.log(buckets) / Math.log(states)));
    }
}

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-11/14-Poor-Pigs.py
class Solution:
    def poorPigs(self, buckets: int, minutesToDie: int, minutesToTest: int) -> int:
        '''
        states ^ pigs >= buckets
        pigs >= log(buckets) / log(states)
        T: O(1) and S: O(1)
        '''
        states = minutesToTest // minutesToDie + 1
        return math.ceil(math.log(buckets) / math.log(states))
